p2p/p2p/src/client_pool.rs

```diff
@@ -127,6 +127,32 @@ impl<N: NetworkZone> ClientPool<N> {
     ) -> impl Iterator<Item = ClientPoolDropGuard<N>> + sealed::Captures<(&'a (), &'b ())> {
         peers.iter().filter_map(|peer| self.borrow_client(peer))
     }
+
+    /// Borrows all [`Client`]s from the pool that have claimed a higher cumulative difficulty than
+    /// the amount passed in.
+    ///
+    /// The [`Client`]s are wrapped in [`ClientPoolDropGuard`] which
+    /// will return the clients to the pool when they are dropped.
+    pub fn clients_with_more_cumulative_difficulty(
```

Comment:
This is the new way for the block downloader to get peers to sync from.

This function is horribly expensive, however in a future PR I have some ideas on how to make it cheaper

Reply:
> Is it expensive because of the Mutex?

Yeah and the `DashMap` which is a bunch of `RwLocks`

> Also, would the below have the same lifetime semantics? I don't think Captures is needed anymore right?

It's not however I would rather keep it as it is as the current way allows more uses.